data/ui_scripts/server_list/serverlist.lua

```diff
@@ -153,6 +336,192 @@ SystemLinkJoinMenu.AddServerButton = function(menu, controller, index)
+			if servers[ip] ~= nil then
+				SystemLinkJoinMenu.AddServerButton(menu, controller, serversSorted[index].index)
+			end 
+		end
+	elseif type == "blacklisted" then
+		menu.removefromblacklist = menu:AddHelp({
+			name = "add_button_helper_text",
+			button_ref = "button_right",
+			helper_text = Engine.Localize("@SERVERLIST_REMOVE_FROM_BLACKLIST"),
+			side = "right",
+			clickable = false,
+			priority = -1000
+		}, nil, nil, true)
+
+		menu.clearlist = menu:AddHelp({
+			name = "add_button_helper_text",
+			button_ref = "button_left_trigger",
+			helper_text = Engine.Localize("@SERVERLIST_CLEAR_LIST"),
+			side = "right",
+			clickable = false,
+			priority = -1000
+		}, function(f21_arg0, f21_arg1)
+			local type = Engine.GetDvarString("ui_server_type")
+			if type == "favourites" then
+				file = io.open("players2/favourites.txt", "w")
+				file:write("")
+				file:close()
+			elseif type == "blacklisted" then
+				file = io.open("players2/blacklisted.txt", "w")
+				file:write("")
+				file:close()
+			end
+			SystemLinkJoinMenu.BuildList(menu, Engine.GetFirstActiveController())
+		end)
+
+		local servers = lines_as_hashmap("players2/blacklisted.txt")
+		for index = 1, #serversSorted, 1 do
+			local ip = Lobby.GetServerData(controller,  serversSorted[index].index, 6)
+			if servers[ip] ~= nil then
+				SystemLinkJoinMenu.AddServerButton(menu, controller, serversSorted[index].index)
+			end 
+		end
+	else
+		menu.addtofavourites = menu:AddHelp({
+			name = "add_button_helper_text",
+			button_ref = "button_left",
+			helper_text = Engine.Localize("@SERVERLIST_ADD_TO_BLACKLIST"),
+			side = "right",
+			clickable = true,
+			priority = -1000
+		}, nil, nil, true)
+
+		menu.addtoblacklist = menu:AddHelp({
+			name = "add_button_helper_text",
+			button_ref = "button_right",
+			helper_text = Engine.Localize("@SERVERLIST_ADD_TO_FAVOURITES"),
+			side = "right",
+			clickable = true,
+			priority = -1000
+		}, nil, nil, true)
+	
+		local servers = lines_as_hashmap("players2/blacklisted.txt")
+		for index = 1, #serversSorted, 1 do
+			local ip = Lobby.GetServerData(controller,  serversSorted[index].index, 6)
+			if servers[ip] == nil then
+				SystemLinkJoinMenu.AddServerButton(menu, controller, serversSorted[index].index)
+			end
+		end
+	end
+	
+	if menu.serverCount > 0 then
+		menu.list:registerEventHandler( LUI.FormatAnimStateFinishEvent( "default" ), function ( element, event )
+			ListPaging.InitList( menu.list, 18, nil, nil, nil, {
+				enabled = false
+			} )
+		end )
+		menu.list:animateToState( "default" )
+	end
+end
+
+SystemLinkJoinMenu.OnJoinGame =  function ( server, menu )
+	local ConnectTo = Lobby.JoinServer
+	local controller = menu.controller
+	if not controller then
+		controller = Engine.GetFirstActiveController()
+	end
+
+	if Lobby.GetServerData(controller, server.index, 5) == "1" then
+		LUI.FlowManager.RequestAddMenu(menu, "server_password_field", false, controller, true,  {
```

Comment:
From what I saw in the image the password prompt doesn't use '*' characters to hide the password

Reply:
Yes, it do not use it because for some reason also with * once you try to set the password again will be visible untile you press on the input field.